CONTRIBUTING.rst

```diff
@@ -25,42 +25,205 @@ Pull Request Checklist
-   tests <#running-unit-tests>`__
+-  Refer to the `Windows users <#windows-users>`__ section if your development environment
+   is Windows.
+
+   - `Creating Virtual Environment <#creating-virtual-environment>`__
+   - `Installing Dependencies <#installing-dependencies>`__
+   - `Running Unit Tests <#windows-running-unit-tests>`__
+   - `Coding Style <#windows-coding-style>`__
+
+-  Refer to the `Linux/Mac users <#linux-and-mac-users>`__ section if your development environment
+   is either Linux or Mac.
+
+   - `Running Unit Tests <#linux-mac-running-unit-tests>`__
+   - `Coding Style <#linux-mac-coding-style>`__
+   - `Test Coverage <#linux-mac-test-coverage>`__
+   - `Generate Docs <#linux-mac-generate-docs>`__
 
 General guidelines for contribution
 ------------------------------------
 
--  Include unit tests when you contribute new features, as they help to
-   a) prove that your code works correctly, and b) guard against future
-   breaking changes to lower the maintenance cost.
+-  Include unit tests when you contribute new features, as they help to:
+   
+   - Prove that your code works correctly.
+   - Guard against future breaking changes to lower the maintenance cost.
+
 -  Bug fixes also generally require unit tests, because the presence of
    bugs usually indicates insufficient test coverage.
+-  Make sure that your code changes complies with the coding style of the
+   project.
 -  Keep API compatibility in mind when you change code. Reviewers of
    your pull request will comment on any API compatibility issues.
 
+Windows Users
+---------------
+For contributors using Windows operating system, you have to manually run
+the following commands since as of now the ``make`` commands works with
+Linux/Mac OS. **BTW, you are open to solve this issue!!**
+
+
+----------------------------
+Creating virtual environment
+----------------------------
+To create a virtual environment in python, run the following command:
+
+.. code:: bash
+
+   python -m venv <name_of_your_virtual_environment>
+
+------------------------
+Installing dependencies
+------------------------
+.. _windows-installing-dependencies:
+
+Once your virtual environment is set up, activate your virtual environment
+by executing the command: ``./venv/Scripts/Activate.ps1``. Then, install
+the dependencies for your virtual environment by running the following command:
+
+- To install development dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\dev-requirements.txt
+
+- To install NLP related dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\nlp-requirements.txt
+
+- To install NLP properties related dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\nlp-prop-requirements.txt
+
+- To install vision development dependencies:
+
+   .. code:: bash
+
+      pip install -r .\requirements\vision-requirements.txt
+
+..
+
+   If you are working with NLP module, then you may require to install
```

Comment:
fasttext doesn't just install from pip on windows? I know that on some machines it requires a specific gcc version or such...

Reply:
Yeah actually on Windows, it is quite a difficult/unclear process. The best and easiest way is that if we have a wheel file, then we can directly install it on windows.